src/assets/locales/README.md

```diff
@@ -9,6 +9,231 @@
+
+Example:
+```jsx
+
+// Good
+function SignInModal() {
+ return <FormattedMessage id="sign-in-modal.complete" defaultMessage="Sign in complete" />;
+}
+
+// Bad
+function SignInModal() {
+ return <FormattedMessage id="sign-in.complete" defaultMessage="Sign in complete" />;
+}
+
+// Good
+function ObjectListItem({ item }) {
+ return (
+  <li>
+    {item.name}
+    <button>
+      <FormattedMessage id="object-list.select" defaultMessage="Select Object" />
+    </button>
+  </li>
+  );
+}
+
+export function ObjectList({ objects }) {
+ return <ul>{objects.map(o => <ObjectListItem item={o} />)}</ul>;
+}
+```
+
+### 3. Always specify a default message.
+Default messages will be used to provide context to translators, they are the fallback message if it is not defined for another language, and they are used as the English translation by default.
+
+```jsx
+// Good
+<FormattedMessage id="sign-in-modal.complete" defaultMessage="Sign in complete" />
+
+// Bad
+<FormattedMessage id={`sign-in-modal.complete`} />
+
+// Good
+const signInCompleteMessage = defineMessage({
+  id: "sign-in-modal.complete",
+  defaultMessage: "Sign in complete"
+});
+
+// Bad
+const signInCompleteMessage = defineMessage({
+  id: "sign-in-modal.complete"
+});
+```
+
+### 4. All strings presented to users should be localized including accessibility features
+It is not enough to localize just the visible text. All labels, alt text, and callouts should also be localized.
+
+```jsx
+<img alt={<FormattedMessage id="logo.alt-text" defaultMessage="Hubs Logo" />} src="logo.png" />
+
+<div aria-label={<FormattedMessage id="element.label" defaultMessage="This is a label" />}></div>
+
+<div title={<FormattedMessage id="element.title" defaultMessage="This is a title" />}></div>
+```
+
+### 5. Use react-intl's formatter for formatting concatenated strings.
+Different languages have different grammatical rules and sentence structure. Relying on string concatenation that works in English will not always make sense in another language. Use react-intl's `values` prop to format messages with variables or rich formatting. If formatting gets too complicated break it up into multiple messages.
+
+```jsx
+// Bad
+<p>
+  <small>
+    By proceeding, you agree to the{" "}
+    {showTerms && (
+      <>
+        <a rel="noopener noreferrer" target="_blank" href={termsUrl}>
+          terms of use
+        </a>{" "}
+      </>
+    )}
+    {showTerms && showPrivacy && "and "}
+    {showPrivacy && (
+      <a rel="noopener noreferrer" target="_blank" href={privacyUrl}>
+        privacy notice
+      </a>
+    )}.
+  </small>
+</p>
+```
+```jsx
+// Good
+export function LegalMessage({ termsUrl, privacyUrl }) {
+  const toslink = useCallback(
```

Comment:
I think the capitalization here is correct as lowercase since it's being used as an HTML element name.

Notice how it's used here:
```jsx
<FormattedMessage
  id="legal-message.privacy"
  defaultMessage="By proceeding, you agree to the <privacylink>privacy notice</privacylink>."
  values={{
    privacylink
  }}
/>
 ```